src/Controls/samples/Controls.Sample/Pages/Controls/LabelPage.xaml.cs

```diff
@@ -1,36 +1,71 @@
-﻿using Microsoft.Maui;
+﻿using System;
+using Microsoft.Maui;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.Graphics;
 
 namespace Maui.Controls.Sample.Pages
 {
 	public partial class LabelPage
 	{
+		readonly Color[] _colors = new Color[] { Colors.Red, Colors.Blue, Colors.Green, Colors.Yellow, Colors.Brown, Colors.Purple, Colors.Orange, Colors.Gray };
+		readonly Random _rand = new Random();
+
 		public LabelPage()
 		{
 			InitializeComponent();
 		}
+
 		void ClickGestureRecognizer_Clicked(System.Object sender, System.EventArgs e)
 		{
-			var rnd = new System.Random();
+			var label = sender as Label;
 
-			var span = sender as Span;
-			if (span != null)
-				span.TextColor = Color.FromRgb((byte)rnd.Next(0, 254), (byte)rnd.Next(0, 254), (byte)rnd.Next(0, 254));
+			if (label == null)
+				return;
+
+			var rnd = new System.Random();
+			GestureSpan.TextColor = Color.FromRgb((byte)rnd.Next(0, 254), (byte)rnd.Next(0, 254), (byte)rnd.Next(0, 254));
 		}
 
 		void ChangeFormattedString_Clicked(object sender, System.EventArgs e)
 		{
 			labelFormattedString.FormattedText = new FormattedString
 			{
 				Spans =
-			{
-				new Span
 				{
-					Text = "Testing"
+					new Span
+					{
+						Text = "Testing"
+					}
 				}
-			}
 			};
 		}
+
+		void OnLink1Tapped(object sender, EventArgs e)
+		{
+			SetRandomBackgroundColor(Link1);
+		}
+
+		void OnLink2Tapped(object sender, EventArgs e)
+		{
+			SetRandomBackgroundColor(Link2);
+		}
+
+		void OnLink3Tapped(object sender, EventArgs e)
+		{
+			SetRandomBackgroundColor(Link3);
+		}
+
+		void SetRandomBackgroundColor(Span span)
+		{
+			var oldColor = span.BackgroundColor;
+
+			Color newColor;
+			do
+			{
+				newColor = _colors[_rand.Next(_colors.Length)];
+			} while (oldColor == newColor);
```

Comment:
I see the use of the do while, but is it really needed? I mean, it just looks like you are doing:
```suggestion
			Color newColor = _colors[_rand.Next(_colors.Length)];
			while (oldColor == newColor) {
			    newColor = _colors[_rand.Next(_colors.Length)];
			}
```
I mention this because if color is a struct we should *be* ok, but if it is a ref type, the compiler is going to ask you to move it to Color? and then we have to use ? everywhere.